autopush/web/base.py

```diff
@@ -0,0 +1,234 @@
+import json
+import time
+import uuid
+from collections import namedtuple
+
+import cyclone.web
+from boto.dynamodb2.exceptions import (
+    ProvisionedThroughputExceededException,
+)
+from twisted.logger import Logger
+from twisted.python import failure
+
+from autopush.db import (
+    hasher,
+    normalize_id,
+)
+from autopush.exceptions import InvalidRequest
+from autopush.router.interface import RouterException
+
+status_codes = {
+    200: "OK",
+    201: "Created",
+    202: "Accepted",
+    400: "Bad Request",
+    401: "Unauthorized",
+    404: "Not Found",
+    413: "Payload Too Large",
+    500: "Internal Server Error",
+    503: "Service Unavailable",
+}
+
+
+class Notification(namedtuple("Notification",
+                   "version data channel_id headers ttl")):
+    """Parsed notification from the request"""
+
+
+class VapidAuthException(Exception):
+    """Exception if the VAPID Auth token fails"""
+    pass
+
+
+class BaseHandler(cyclone.web.RequestHandler):
+    """Common overrides for Push web API's"""
+    cors_methods = ""
+    cors_request_headers = []
+    cors_response_headers = []
+
+    log = Logger()
+
+    #############################################################
+    #                    Cyclone API Methods
+    #############################################################
+    def initialize(self, ap_settings):
+        """Setup basic aliases and attributes"""
+        self.uaid_hash = ""
+        self._uaid = ""
+        self._chid = ""
+        self.start_time = time.time()
+        self.ap_settings = ap_settings
+        self.metrics = ap_settings.metrics
+        self.request_id = str(uuid.uuid4())
+        self._client_info = self._init_info()
+
+    def prepare(self):
+        """Common request preparation"""
+        if self.ap_settings.cors:
+            self.set_header("Access-Control-Allow-Origin", "*")
+            self.set_header("Access-Control-Allow-Methods",
+                            self.cors_methods)
+            self.set_header("Access-Control-Allow-Headers",
+                            ",".join(self.cors_request_headers))
+            self.set_header("Access-Control-Expose-Headers",
+                            ",".join(self.cors_response_headers))
+
+    def write_error(self, code, **kwargs):
```

Comment:
The ErrorLogger is redundant with the new BaseHandler, yup. So the WebPush refactor to this setup will only inherit BaseHandler.